addons/account/tests/test_account_tax.py

```diff
@@ -31,6 +63,133 @@ def test_changing_tax_company(self):
+        self.flush_tracking()
+        self.assertTrue(self.company_data['default_tax_sale'].is_used)
+
+        self.company_data['default_tax_sale'].write({
+            'invoice_repartition_line_ids': [
+                Command.create({'repartition_type': 'tax'}),
+            ],
+            'refund_repartition_line_ids': [
+                Command.create({'repartition_type': 'tax'}),
+            ],
+        })
+        self.flush_tracking()
+
+        previews = self.company_data['default_tax_sale'].message_ids.mapped('preview')
+        self.assertIn(
+            "New Invoice repartition line 4: 100.0 (Factor Percent) None (Account) None (Tax Grids) False (Use in tax closing)",
+            previews
+        )
+        self.assertIn(
+            "New Refund repartition line 4: 100.0 (Factor Percent) None (Account) None (Tax Grids) False (Use in tax closing)",
+            previews
+        )
+
+    def test_logging_of_repartition_lines_update_when_tax_is_used(self):
+        ''' Updating repartition lines in a used tax should be logged. '''
+
+        self.flush_tracking()
+        self.assertTrue(self.company_data['default_tax_sale'].is_used)
+
+        last_invoice_rep_line = self.company_data['default_tax_sale'].invoice_repartition_line_ids.sorted(key=lambda r: r.sequence)[-1]
+        last_refund_rep_line = self.company_data['default_tax_sale'].refund_repartition_line_ids.sorted(key=lambda r: r.sequence)[-1]
+
+        self.company_data['default_tax_sale'].write({
+            "invoice_repartition_line_ids": [
+                Command.update(last_invoice_rep_line.id, {
+                    'factor_percent': -50,
+                    'tag_ids': [Command.create({'name': 'TaxTag12345'})]
+                }),
+            ],
+            "refund_repartition_line_ids": [
+                Command.update(last_refund_rep_line.id, {
+                    'factor_percent': -50,
+                    'account_id': self.company_data['default_account_tax_purchase'].id,
+                }),
+            ],
+        })
+        self.flush_tracking()
+
+        previews = self.company_data['default_tax_sale'].message_ids.mapped('preview')
+        self.assertIn("Invoice repartition line 3: 100.0 -50.0 (Factor Percent) None ['TaxTag12345'] (Tax Grids)", previews)
+        self.assertIn("Refund repartition line 3: 100.0 -50.0 (Factor Percent) None 131000 Tax Paid (Account) False True (Use in tax closing)", previews)
+
+    def test_logging_of_repartition_lines_reordering_when_tax_is_used(self):
+        ''' Reordering repartition lines in a used tax should be logged. '''
+
+        self.flush_tracking()
+        self.assertTrue(self.company_data['default_tax_sale'].is_used)
+
+        last_invoice_rep_line = self.company_data['default_tax_sale'].invoice_repartition_line_ids.sorted(key=lambda r: r.sequence)[-1]
+        before_last_invoice_rep_line = self.company_data['default_tax_sale'].invoice_repartition_line_ids.sorted(key=lambda r: r.sequence)[-2]
```

Comment:
```suggestion
        sorted_invoice_rep_lines = self.company_data['default_tax_sale'].invoice_repartition_line_ids.sorted(key=lambda r: r.sequence)
        last_invoice_rep_line, before_last_invoice_rep_line = sorted_invoice_rep_lines[-2:]
```
I know it's just a test, but no need to sort twice :wink: